src/coreclr/jit/gentree.cpp

```diff
@@ -17950,6 +17950,38 @@ bool Compiler::gtIsStaticGCBaseHelperCall(GenTree* tree)
     return false;
 }
 
+//------------------------------------------------------------------------
+// gtIsTypeof: Checks if the tree is a typeof()
+//
+// Arguments:
+//    tree   - the tree that is checked
+//    handle - (optional, default nullptr) - if non-null is set to the type
+//
+// Return Value:
+//    Is the tree typeof()
+//
+bool Compiler::gtIsTypeof(GenTree* tree, CORINFO_CLASS_HANDLE* handle)
```

Comment:
Seems like you should try and leverage `gtGetTypeProducerKind` and/or `gtIsTypeHandleToRuntimeTypeHelper` here.

Reply:
`gtGetTypeProducerKind` doesn't seem particularly useful here since we don't know the exact type with .GetType() and we don't want to return null types here so I've used `gtIsTypeHandleToRuntimeTypeHelper` and added tests for NRE with `Type.GetTypeFromHandle(default)`.